temporal-sdk/src/main/java/io/temporal/worker/WorkerOptions.java

```diff
@@ -349,6 +354,22 @@ public Builder setBuildId(String buildId) {
       return this;
     }
 
+    /**
+     * During graceful shutdown, like calling {@link WorkerFactory#shutdown()}, if "sticky worker"
+     * is enabled, this timeout controls how long to wait for the sticky task queue to drain before
+     * shutting down the worker. If set the worker will stop making new poll requests on the normal
+     * task queue, but will continue to poll the sticky task queue until the timeout is reached.
+     * This value should always be greater than clients rpc long poll timeout, which can be set via
+     * {@link WorkflowServiceStubsOptions.Builder#setRpcLongPollTimeout(Duration)}.
```

Comment:
Now I wonder if it'd make more sense to make this a `boolean` for `waitForStickyEmptyPoll`, because what if you have a time race? For instance, what if my long poll timeout and my sticky drain timeout is 1m. And as soon as the poll comes back empty I haven't hit sticky drain timeout, so I make another sticky poll for just some nanoseconds before sticky drani timeout does hit? Maybe that's ok though.

Reply:
That poll would be cancelled, there is a possible race, but the race also exists for normal shutdown as well and I don't think we should tackle it here.

Reply:
:+1: Would like to see us not kill poll calls anywhere, but yes can be tackled elsewhere